File: lib/sample/src/greeting.cc
#include "sample_cpp/greeting.h"

#include <string>

namespace sample_cpp {

std::string greeting(std::string_view name) {
    if (name.empty()) {
        return "Hello, world!";
    }

    return "Hello, " + std::string(name) + "!";
}

}  // namespace sample_cpp

File: components/sample/src/main/java/com/example/monorepo/samplejava/Greeting.java
package com.example.monorepo.samplejava;

public final class Greeting {
  private Greeting() {}

  public static String forName(String name) {
    String normalized = name == null ? "" : name.trim();
    if (normalized.isEmpty()) {
      normalized = "monorepo";
    }
    return "Hello, " + normalized + " from Java";
  }
}
